src/context/VernaContextProvider.tsx

```diff
@@ -136,7 +99,7 @@ function VernaContextProvider({
     getSelectedUiSchema(defaultUiSchema, defaultSelector),
   );
   const [formData, setFormData] = useState(getDefaultValues(defaultFormValues, defaultSelector));
-  const [readOnly, setReadOnly] = useState<boolean>(defaultReadOnly);
+  const [_isEditorMode, setIsEditorMode] = useState<boolean>(isEditorMode);
```

Comment:
Why not create a `isPreview` state? In this way, we are able to enforce the preview mode internally but keep editorMode state delegated to the upper scope.